### ISSUE TITLE
qt/5.12.6: unable compile with clang-10

### ISSUE THREAD

**Issue by yssource** [bincrafters/community#1185]
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **zlib/1.2.11**
  * Operating System+version: **Linux Ubuntu 18.04**
  * Compiler+version: **Clang 10**
  * Conan version: **conan 1.18.0**
  * Python version: **Python 3.7.7+**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=clang
compiler.version=10
compiler.libcxx=libstdc++11
build_type=Debug
[options]
qt:qtdeclarative=True
qt:qttools=True
qt:qtwebsockets=True
[build_requires]
[env]
CC=/usr/bin/clang
CXX=/usr/bin/clang++
CONAN_CMAKE_GENERATOR=Ninja
```

### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=clang
compiler.libcxx=libstdc++11
compiler.version=10
os=Linux
os_build=Linux
[options]
qt:qtdeclarative=True
qt:qttools=True
qt:qtwebsockets=True
[build_requires]
[env]
CC=/usr/bin/clang
CONAN_CMAKE_GENERATOR=Ninja
CXX=/usr/bin/clang++
conanfile.py: Installing package
Requirements
    bzip2/1.0.8 from 'conan-center' - Cache
    dataframe/1.7.0 from local cache - Cache
    double-conversion/3.1.5 from 'conan-center' - Cache
    expat/2.2.9 from 'conan-center' - Cache
    fontconfig/2.13.91@conan/stable from 'conan-center' - Cache
    freetype/2.10.0 from 'conan-center' - Cache
    glib/2.58.3@bincrafters/stable from 'conan-center' - Cache
    harfbuzz/2.6.2@bincrafters/stable from 'conan-center' - Cache
    icu/64.2 from 'conan-center' - Cache
    libelf/0.8.13 from 'conan-center' - Cache
    libffi/3.2.1 from 'conan-center' - Cache
    libiconv/1.15 from 'conan-center' - Cache
    libjpeg/9c from 'conan-center' - Cache
    libmount/2.33.1 from 'conan-center' - Cache
    libmysqlclient/8.0.17 from 'conan-center' - Cache
    libpng/1.6.37 from 'conan-center' - Cache
    libpq/11.5 from 'conan-center' - Cache
    libpthread-stubs/0.1@bincrafters/stable from 'bincrafters' - Cache
    libselinux/2.9@bincrafters/stable from 'bincrafters' - Cache
    libuuid/1.0.3 from 'conan-center' - Cache
    libx11/1.6.8@bincrafters/stable from 'bincrafters' - Cache
    libxau/1.0.9@bincrafters/stable from 'bincrafters' - Cache
    libxcb/1.13.1@bincrafters/stable from 'bincrafters' - Cache
    libxdmcp/1.1.3@bincrafters/stable from 'bincrafters' - Cache
    nlohmann_json/3.7.3 from 'conan-center' - Cache
    odbc/2.3.7 from 'conan-center' - Cache
    openssl/1.1.1d from 'conan-center' - Cache
    pcre/8.41 from 'conan-center' - Cache
    pcre2/10.33 from 'conan-center' - Cache
    qt/5.12.6@bincrafters/stable from 'bincrafters' - Cache
    sqlite3/3.29.0 from 'conan-center' - Cache
    util-macros/1.19.2@bincrafters/stable from 'bincrafters' - Cache
    xcb-proto/1.13@bincrafters/stable from 'bincrafters' - Cache
    xkbcommon/0.8.4@bincrafters/stable from 'bincrafters' - Cache
    xkeyboard-config/2.28@bincrafters/stable from 'bincrafters' - Cache
    xorgproto/2019.1@bincrafters/stable from 'bincrafters' - Cache
    xproto/7.0.31@bincrafters/stable from 'bincrafters' - Cache
    xsimd/7.4.6 from local cache - Cache
    xtensor/0.21.3 from local cache - Cache
    xtl/0.6.12 from 'conan-center' - Cache
    xtrans/1.4.0@bincrafters/stable from 'bincrafters' - Cache
    zlib/1.2.11 from 'conan-center' - Cache
Packages
    bzip2/1.0.8:23219b68e873594c97b915e5b682fd61d2caca86 - Cache
    dataframe/1.7.0:803d1cd04e225b70657161961aaf49e6f4674680 - Cache
    double-conversion/3.1.5:803d1cd04e225b70657161961aaf49e6f4674680 - Cache
    expat/2.2.9:72c3493baf37f6fd0e9e64ab36754d5a2f6479a9 - Cache
    fontconfig/2.13.91@conan/stable:2d1611e33c9e5e8f555861b7a20e8ba0edb65718 - Cache
    freetype/2.10.0:3b6447caddb0e07baf31f99b0c46405a4818b04e - Cache
    glib/2.58.3@bincrafters/stable:b0a0dfe6da954598a8a0f968801c2159e0912a4a - Cache
    harfbuzz/2.6.2@bincrafters/stable:8a27ec6a0223c92d7e7c7834ab9408dcf747abea - Cache
    icu/64.2:a37e2cd09441100a2dd9b7449a425c390d8db4e4 - Cache
    libelf/0.8.13:72c3493baf37f6fd0e9e64ab36754d5a2f6479a9 - Cache
    libffi/3.2.1:72c3493baf37f6fd0e9e64ab36754d5a2f6479a9 - Cache
    libiconv/1.15:72c3493baf37f6fd0e9e64ab36754d5a2f6479a9 - Cache
    libjpeg/9c:72c3493baf37f6fd0e9e64ab36754d5a2f6479a9 - Cache
    libmount/2.33.1:72c3493baf37f6fd0e9e64ab36754d5a2f6479a9 - Cache
    libmysqlclient/8.0.17:c4dee131c62514ba465188ebc88e2dabb980a75b - Cache
    libpng/1.6.37:4ada6b9b4eb17cabe936d1e8dc720431e269a9a3 - Cache
    libpq/11.5:48d527d4d92b5d8c82e971e87e71aae884a9e873 - Cache
    libpthread-stubs/0.1@bincrafters/stable:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    libselinux/2.9@bincrafters/stable:32777db2abef0c6300f8d11778a6c0e35d5b9f28 - Cache
    libuuid/1.0.3:72c3493baf37f6fd0e9e64ab36754d5a2f6479a9 - Cache
    libx11/1.6.8@bincrafters/stable:2512f44babcc236f7825cbe95bd1dd41f503ffc7 - Cache
    libxau/1.0.9@bincrafters/stable:82c7a1154e4e6285931adf51f3342bac778307e8 - Cache
    libxcb/1.13.1@bincrafters/stable:a5c9e96e5e4451a7bfae67296ec18cb3b4c55ece - Cache
    libxdmcp/1.1.3@bincrafters/stable:9a769642b540d98a8fa75e1ef9deb96f121e6b93 - Cache
    nlohmann_json/3.7.3:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    odbc/2.3.7:efec58ec2be55c7193cb969d8dccf25e850d18f3 - Cache
    openssl/1.1.1d:72c3493baf37f6fd0e9e64ab36754d5a2f6479a9 - Cache
    pcre/8.41:48bc3a3d1d15fb3ad0eef9e37a298688d4ca4679 - Cache
    pcre2/10.33:3071444d1594ddd98df6f12fb3ce3a93129e3c76 - Cache
    qt/5.12.6@bincrafters/stable:a2b96327c1be568174824a35e2a0ca3abb804453 - Build
    sqlite3/3.29.0:06a7307c9e97c46edd29198dd04c94b1b4f2a225 - Cache
    util-macros/1.19.2@bincrafters/stable:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    xcb-proto/1.13@bincrafters/stable:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    xkbcommon/0.8.4@bincrafters/stable:b1e5bb42d80e30d0c504696714177fafee979031 - Cache
    xkeyboard-config/2.28@bincrafters/stable:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    xorgproto/2019.1@bincrafters/stable:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    xproto/7.0.31@bincrafters/stable:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    xsimd/7.4.6:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    xtensor/0.21.3:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    xtl/0.6.12:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    xtrans/1.4.0@bincrafters/stable:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    zlib/1.2.11:72c3493baf37f6fd0e9e64ab36754d5a2f6479a9 - Cache

qt/5.12.6@bincrafters/stable: WARN: Build folder is dirty, removing it: /home/jimmy/.conan/data/qt/5.12.6/bincrafters/stable/build/a2b96327c1be568174824a35e2a0ca3abb804453
qt/5.12.6@bincrafters/stable: ERROR: Package 'a2b96327c1be568174824a35e2a0ca3abb804453' build failed
qt/5.12.6@bincrafters/stable: WARN: Build folder /home/jimmy/.conan/data/qt/5.12.6/bincrafters/stable/build/a2b96327c1be568174824a35e2a0ca3abb804453
ERROR: qt/5.12.6@bincrafters/stable: Error in build() method, line 590
	self.run("%s/qt5/configure %s" % (self.source_folder, " ".join(args)))
	ConanException: Error 3 while executing /home/jimmy/.conan/data/qt/5.12.6/bincrafters/stable/source/qt5/configure -shared -confirm-license -silent -nomake examples -nomake tests -prefix /home/jimmy/.conan/data/qt/5.12.6/bincrafters/stable/package/a2b96327c1be568174824a35e2a0ca3abb804453 -opensource -debug -skip qtsvg -skip qtactiveqt -skip qtscript -skip qtmultimedia -skip qtxmlpatterns -skip qttranslations -skip qtdoc -skip qtlocation -skip qtsensors -skip qtconnectivity -skip qtwayland -skip qt3d -skip qtimageformats -skip qtgraphicaleffects -skip qtquickcontrols -skip qtserialbus -skip qtserialport -skip qtx11extras -skip qtmacextras -skip qtwinextras -skip qtandroidextras -skip qtwebchannel -skip qtwebengine -skip qtcanvas3d -skip qtwebview -skip qtquickcontrols2 -skip qtpurchasing -skip qtcharts -skip qtdatavis3d -skip qtvirtualkeyboard -skip qtgamepad -skip qtscxml -skip qtspeech -skip qtnetworkauth -skip qtremoteobjects -skip qtwebglplugin --zlib=system -opengl desktop -openssl-linked --glib=yes --pcre=system --fontconfig=yes --icu=yes --sql-mysql=yes --sql-psql=yes --sql-odbc=yes -system-doubleconversion -system-freetype -system-harfbuzz -system-libjpeg -system-libpng -system-sqlite -I /home/jimmy/.conan/data/zlib/1.2.11/_/_/package/72c3493baf37f6fd0e9e64ab36754d5a2f6479a9/include "ZLIB_LIBS=-lz" -I /home/jimmy/.conan/data/openssl/1.1.1d/_/_/package/72c3493baf37f6fd0e9e64ab36754d5a2f6479a9/include "OPENSSL_LIBS=-lssl -lcrypto" -I /home/jimmy/.conan/data/pcre2/10.33/_/_/package/3071444d1594ddd98df6f12fb3ce3a93129e3c76/include -D PCRE2_STATIC "PCRE2_LIBS=-lpcre2-posix -lpcre2-8 -lpcre2-16 -lpcre2-32 -lz -lbz2" -I /home/jimmy/.conan/data/glib/2.58.3/bincrafters/stable/package/b0a0dfe6da954598a8a0f968801c2159e0912a4a/include -I /home/jimmy/.conan/data/glib/2.58.3/bincrafters/stable/package/b0a0dfe6da954598a8a0f968801c2159e0912a4a/include/glib-2.0 -I /home/jimmy/.conan/data/glib/2.58.3/bincrafters/stable/package/b0a0dfe6da954598a8a0f968801c2159e0912a4a/lib/glib-2.0/include "GLIB_LIBS=-lgio-2.0 -lgmodule-2.0 -lgobject-2.0 -lgthread-2.0 -lglib-2.0 -lz -lffi -lpcreposix -lpcre -lbz2 -lelf -lmount -lblkid -lselinux -lsepol -lpcre2-posix -lpcre2-8 -lpcre2-16 -lpcre2-32" -I /home/jimmy/.conan/data/double-conversion/3.1.5/_/_/package/803d1cd04e225b70657161961aaf49e6f4674680/include "DOUBLECONVERSION_LIBS=-ldouble-conversion" "FREETYPE_INCDIR=/home/jimmy/.conan/data/freetype/2.10.0/_/_/package/3b6447caddb0e07baf31f99b0c46405a4818b04e/include/freetype2" "FREETYPE_LIBS=-lfreetyped -lm -lpng16d -lz -lbz2" -I /home/jimmy/.conan/data/fontconfig/2.13.91/conan/stable/package/2d1611e33c9e5e8f555861b7a20e8ba0edb65718/include "FONTCONFIG_LIBS=-lfontconfig -lm -lpthread -lfreetyped -lpng16d -lz -lbz2 -lexpat -luuid" -I /home/jimmy/.conan/data/icu/64.2/_/_/package/a37e2cd09441100a2dd9b7449a425c390d8db4e4/include -D U_STATIC_IMPLEMENTATION "ICU_LIBS=-licui18n -licuio -licutest -licutu -licuuc -licudata -ldl" -I /home/jimmy/.conan/data/harfbuzz/2.6.2/bincrafters/stable/package/8a27ec6a0223c92d7e7c7834ab9408dcf747abea/include -I /home/jimmy/.conan/data/harfbuzz/2.6.2/bincrafters/stable/package/8a27ec6a0223c92d7e7c7834ab9408dcf747abea/include/harfbuzz "HARFBUZZ_LIBS=-lharfbuzz -lfreetyped -lm -lpng16d -lz -lbz2 -lgio-2.0 -lgmodule-2.0 -lgobject-2.0 -lgthread-2.0 -lglib-2.0 -lffi -lpcreposix -lpcre -lelf -lmount -lblkid -lselinux -lsepol -lpcre2-posix -lpcre2-8 -lpcre2-16 -lpcre2-32" -I /home/jimmy/.conan/data/libjpeg/9c/_/_/package/72c3493baf37f6fd0e9e64ab36754d5a2f6479a9/include -D LIBJPEG_STATIC "LIBJPEG_LIBS=-ljpeg" -I /home/jimmy/.conan/data/libpng/1.6.37/_/_/package/4ada6b9b4eb17cabe936d1e8dc720431e269a9a3/include "LIBPNG_LIBS=-lpng16d -lm -lz" -I /home/jimmy/.conan/data/sqlite3/3.29.0/_/_/package/06a7307c9e97c46edd29198dd04c94b1b4f2a225/include "SQLITE_LIBS=-lsqlite3" -I /home/jimmy/.conan/data/libmysqlclient/8.0.17/_/_/package/c4dee131c62514ba465188ebc88e2dabb980a75b/include "MYSQL_LIBS=-lmysqlclient -lssl -lcrypto -lz" -I /home/jimmy/.conan/data/libpq/11.5/_/_/package/48d527d4d92b5d8c82e971e87e71aae884a9e873/include "PSQL_LIBS=-lpgcommon -lpq -lz" -I /home/jimmy/.conan/data/odbc/2.3.7/_/_/package/efec58ec2be55c7193cb969d8dccf25e850d18f3/include "ODBC_LIBS=-lodbc -lodbccr -lodbcinst -lltdl -ldl -liconv" -L /home/jimmy/.conan/data/zlib/1.2.11/_/_/package/72c3493baf37f6fd0e9e64ab36754d5a2f6479a9/lib -L /home/jimmy/.conan/data/openssl/1.1.1d/_/_/package/72c3493baf37f6fd0e9e64ab36754d5a2f6479a9/lib -L /home/jimmy/.conan/data/pcre2/10.33/_/_/package/3071444d1594ddd98df6f12fb3ce3a93129e3c76/lib -L /home/jimmy/.conan/data/bzip2/1.0.8/_/_/package/23219b68e873594c97b915e5b682fd61d2caca86/lib -L /home/jimmy/.conan/data/glib/2.58.3/bincrafters/stable/package/b0a0dfe6da954598a8a0f968801c2159e0912a4a/lib -L /home/jimmy/.conan/data/libffi/3.2.1/_/_/package/72c3493baf37f6fd0e9e64ab36754d5a2f6479a9/lib -L /home/jimmy/.conan/data/pcre/8.41/_/_/package/48bc3a3d1d15fb3ad0eef9e37a298688d4ca4679/lib -L /home/jimmy/.conan/data/libelf/0.8.13/_/_/package/72c3493baf37f6fd0e9e64ab36754d5a2f6479a9/lib -L /home/jimmy/.conan/data/libmount/2.33.1/_/_/package/72c3493baf37f6fd0e9e64ab36754d5a2f6479a9/lib -L /home/jimmy/.conan/data/libselinux/2.9/bincrafters/stable/package/32777db2abef0c6300f8d11778a6c0e35d5b9f28/lib -L /home/jimmy/.conan/data/double-conversion/3.1.5/_/_/package/803d1cd04e225b70657161961aaf49e6f4674680/lib -L /home/jimmy/.conan/data/freetype/2.10.0/_/_/package/3b6447caddb0e07baf31f99b0c46405a4818b04e/lib -L /home/jimmy/.conan/data/libpng/1.6.37/_/_/package/4ada6b9b4eb17cabe936d1e8dc720431e269a9a3/lib -L /home/jimmy/.conan/data/fontconfig/2.13.91/conan/stable/package/2d1611e33c9e5e8f555861b7a20e8ba0edb65718/lib -L /home/jimmy/.conan/data/expat/2.2.9/_/_/package/72c3493baf37f6fd0e9e64ab36754d5a2f6479a9/lib -L /home/jimmy/.conan/data/libuuid/1.0.3/_/_/package/72c3493baf37f6fd0e9e64ab36754d5a2f6479a9/lib -L /home/jimmy/.conan/data/icu/64.2/_/_/package/a37e2cd09441100a2dd9b7449a425c390d8db4e4/lib -L /home/jimmy/.conan/data/harfbuzz/2.6.2/bincrafters/stable/package/8a27ec6a0223c92d7e7c7834ab9408dcf747abea/lib -L /home/jimmy/.conan/data/libjpeg/9c/_/_/package/72c3493baf37f6fd0e9e64ab36754d5a2f6479a9/lib -L /home/jimmy/.conan/data/sqlite3/3.29.0/_/_/package/06a7307c9e97c46edd29198dd04c94b1b4f2a225/lib -L /home/jimmy/.conan/data/libmysqlclient/8.0.17/_/_/package/c4dee131c62514ba465188ebc88e2dabb980a75b/lib -L /home/jimmy/.conan/data/libpq/11.5/_/_/package/48d527d4d92b5d8c82e971e87e71aae884a9e873/lib -L /home/jimmy/.conan/data/odbc/2.3.7/_/_/package/efec58ec2be55c7193cb969d8dccf25e850d18f3/lib -L /home/jimmy/.conan/data/libiconv/1.15/_/_/package/72c3493baf37f6fd0e9e64ab36754d5a2f6479a9/lib -mysql_config /home/jimmy/.conan/data/libmysqlclient/8.0.17/_/_/package/c4dee131c62514ba465188ebc88e2dabb980a75b/bin/mysql_config -psql_config /home/jimmy/.conan/data/libpq/11.5/_/_/package/48d527d4d92b5d8c82e971e87e71aae884a9e873/bin/pg_config -qt-xcb -platform linux-clang QMAKE_CC="/usr/bin/clang" QMAKE_LINK_C="/usr/bin/clang" QMAKE_LINK_C_SHLIB="/usr/bin/clang" QMAKE_CXX="/usr/bin/clang++" QMAKE_LINK="/usr/bin/clang++" QMAKE_LINK_SHLIB="/usr/bin/clang++" QMAKE_CXXFLAGS+="-ftemplate-depth=1024"

```

</details>

**Comment by Cscotttaakan** [comment]
Try removing the cache, doing a full clean install and post the log
`conan remove * -f`

**Comment by yssource** [comment]
This problem happens when I upgraded my clang9 to clang-10.

Actually I have removed the 5.12.6 cache, 
`rm -rfv ~/.conan/data/qt/5.12.6/`

 I installed a lot of packages.
Do I need to remove all the conan cache?

`conan install qt/5.12.6@bincrafters/stable --build missing` # fail to compile
`conan install qt/5.14.2@bincrafters/stable --build missing` # successfully compiled.

the log is too large to fail to upload to the github issue.

**Comment by Cscotttaakan** [comment]
I think just removing qt is fine, but you'll have to post a log of the qt portion to be able to see what's wrong with it.

**Comment by ericLemanissier** [comment]
We really need the full log to help you, you can use any service (like https://gist.github.com/ for example) if it does not fit in a github message.

**Comment by yssource** [comment]
Thanks, a lot.

https://gist.githubusercontent.com/yssource/f09507f7ab090fc99a7b2aa1941ebf25/raw/ffb2d00233de66f6fe243a7eea018d4916849a9d/qt_5.12.6_clang10.sh

**Comment by yssource** [comment]
another test log.

https://gist.github.com/yssource/b397670bd1f56de18819d190127e9b79/raw/80e7a710f0c44efd18e864a80281b70791e667f5/qt_5.12.6_clang10_test2.log

**Comment by Cscotttaakan** [comment]
Are these the full logs? It seems partial since it doesnt say specifically conan fails at build.

**Comment by ericLemanissier** [comment]
Qt fails to detect openssl, harfbuzz and mysql, because it ignores the pthread flag(because of a qt bug). It should be solved by using 5.12.8

**Comment by yssource** [comment]
@ericLemanissier Thank you.
5.12.8 woks well with clang-10 and my profile.
